enhancements/microshift/etcd-supportability.md

```diff
@@ -0,0 +1,249 @@
+---
+title: etcd-supportability
+authors:
+  - "@dhellmann"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@dusk125, etcd team"
+  - "@pmtk, MicroShift team"
+  - "@vwalek, Support team"
+approvers:
+  - "@pacevedom"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2023-03-02
+last-updated: 2023-03-02
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com//browse/ETCD-356
+see-also:
+  - "/enhancements/microshift/etcd-as-a-transient-systemd-unit.md"
+  - "/enhancements/microshift/periodic-etcd-defragmentation-in-microshift.md"
+  - "/enhancements/microshift/update-and-rollback.md"
+replaces:
+  - "https://github.com/openshift/enhancements/pull/1356"
+---
+
+# Supporting etcd as used in MicroShift
+
+## Summary
+
+This enhancement captures some decisions about how we will provide
+support procedures and tools for etcd as it is configured and used in
+MicroShift deployements.
+
+## Motivation
+
+We need to decide what approach we will take for support procedures
+and tools for the etcd database used by MicroShift. We need to
+consider how closely those procedures should be like the procedures
+for OCP, and whether it makes sense to try to reuse the same tools. We
+have generally taken the stance that we want to align, but not at the
+expense of the user experience or the ability to meet the use cases
+for which MicroShift is being developed.
+
+### User Stories
+
+* As a MicroShift administrator, I want to back up the database used
+  by MicroShift so that I can restore it if something goes wrong with
+  the host.
+* As a MicroShift administrator, I want to restore the MicroShift
+  database when something goes wrong to reset the host to a known good
+  state.
+* As a MicroShift administrator, I want to fix the certificates used
+  to communicate with the database in case something goes wrong.
+
+### Goals
+
+* Describe the relationship between MicroShift and etcd and how it
+  differs from OpenShift.
+* Outline the support procedures for the user stories listed above.
+
+### Non-Goals
+
+* This document will not contain complete support procedures. More
+  complete documentation will be written separately.
+* MicroShift does not own the workloads running on it. Backing up and
```

Comment:
Procedure should specify the application requires a tear down before restoring, right?

Reply:
The process for backup and restore of the application depends on the app. That's why it's out of scope for this document.

Reply:
It would be good for future to include the MTC for workload backups - but MTC heavily relies on the storage providers, so it might need some tweaking. But I agree, that's not scope of MicroShift.